### ISSUE TITLE
[Bug]: Radio with redirect URL fail to play

### ISSUE THREAD

**Issue by akelge** [navidrome/navidrome#2267]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

v0.49.3

### Current Behavior

I added some radios, but one is not playing, it is the only one that requires a 302 response.
The base URL is http://kissfm.kissfmradio.cires21.com/kissfm.mp3, a GET from this URL return a 301 (hint that this is an outdated URL) to https://adhandler.kissfmradio.cires21.com/get_link?url=https://bbkissfm.kissfmradio.cires21.com/bbkissfm.mp3&gdpr=0&gdpr_consent=false that in turn returns a 302 to the final (and temporary, due to the `wmsAuthSign` hash) URL https://bbkissfm.kissfmradio.cires21.com/bbkissfm.mp3?wmsAuthSign=c2VydmVyX3RpbWU9MDMvMjAvMjAyMyAwNDozNDoxOSBQTSZoYXNoX3ZhbHVlPWZ4WXhPdW1zdWljbzcrSjhYa1V1dXc9PSZ2YWxpZG1pbnV0ZXM9MTQ0MCZpZD05MTc3NzQ1Mg==

Using either the first or  the second URL the player hangs with a spinning wheel and no sound.
Nothing in the logs, even at `debug` level.

### Expected Behavior

The radio station starts playing

### Steps To Reproduce

1. add a radio station with URL https://adhandler.kissfmradio.cires21.com/get_link?url=https://bbkissfm.kissfmradio.cires21.com/bbkissfm.mp3&gdpr=0&gdpr_consent=false 
2. try to play the radio station
3. hang

### Environment

```markdown
- OS: docker image
- Browser: google chrome
```

### How Navidrome is installed?

Docker

### Configuration

```toml
navidrome:
    image: deluan/navidrome:0.49.3
    user: 1000:100
    networks:
      - service
    environment:
      ND_SCANSCHEDULE: 0
      ND_LOGLEVEL: debug
      ND_SESSIONTIMEOUT: 24h
      ND_BASEURL: ""
      ND_ENABLEGRAVATAR: "true"
      ND_LASTFM_APIKEY: 46583477c9719db03b8265a1902fce43
      ND_LASTFM_SECRET: e7dce8e68c681c0bb74aebef500d34b6
      ND_SPOTIFY_ID: 5b4e43d68b2243c2853539fbf924de48
      ND_SPOTIFY_SECRET: 063ead1811f44e2f99d3071eb31e257e
      ND_MUSICFOLDER: /music
    volumes:
      - /docker/storage/navidrome:/data
      - /srv/dev-disk-by-label-media/media/Audio/music:/music:ro
      - /srv/dev-disk-by-label-media/media/Audio/music:/shared/media/Audio/music:ro
```

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by kgarner7** [comment]
Hi there! Author of the radio feature here.
This problem is because the endpoint http://kissfm.kissfmradio.cires21.com/kissfm.mp3 does not expose any CORS headers, which means that when your browser tries to fetch it, it will be blocked (check inspect element to see this happening). Unfortunately there's no workaround as-is, but #2187 should ultimately resolve it

**Comment by akelge** [comment]
Oh, nice.
BTW, even using https://adhandler.kissfmradio.cires21.com/get_link?url=https://bbkissfm.kissfmradio.cires21.com/bbkissfm.mp3, i.e. the second link (that is enough to get the token required to access the real streaming URL) does not work, although it works using `play:Sub` IOS client, connected to Navidrome.

Anyways, thanks for the answer, I will wait for the PR to be merged

**Comment by kgarner7** [comment]
Just to clarify one thing, when you play a radio stream on play:sub, it is actually done through the app itself (again, bypassing other errors). 
If you want a live demo of this, it's also available at https://demo.kendallgarner.com (credentials same as Navidrome demo). If you have thoughts about it, feel free to add on the PR as well!

**Comment by akelge** [comment]
Will be solved by #2187
